### PR TITLE
chore(tests): Cleanup kernel shutdown specs

### PR DESCRIPTION
**Pull request by lgeiger**
Use `mock-require`

### CODE REVIEW

File: test/setup.js [hunk 1]
```diff
@@ -140,3 +141,7 @@ mock('react-notification-system', function () {
 mock('spawnteract', {
   'launch': function(kernelSpec, config) { return new Promise() },
 });
+
+mock('fs', {
+  unlinkSync: sinon.spy(),
```
Comment (rgbkrk) [inline test/setup.js:146]: The only problem with this is that it's a global spy, which means that once anyone calls it "calledWith" will be true.

File: test/renderer/api/kernel-spec.js [hunk 2]
```diff
@@ -68,19 +65,15 @@ describe('forceShutdownKernel', () => {
     expect(kernel.spawn.kill).to.have.been.calledWith('SIGKILL');
```
Comment (rgbkrk) [inline test/renderer/api/kernel-spec.js:66]: I'd set up the spy in each of these functions that need it (still making the mock require in the other file) by doing

spy = sinon.spy(fs, 'unlinkSync')

You can then disattach the spy at the end of this test.

**Review by rgbkrk — CHANGES_REQUESTED** [review]

**Comment by rgbkrk** [comment]
Thanks for cleaning this one up.

File: test/renderer/api/kernel-spec.js (hunk 2 above)
Comment (lgeiger) [reply test/renderer/api/kernel-spec.js:66]: That's what I tried first but I can't get it to work.

Comment (rgbkrk) [reply test/renderer/api/kernel-spec.js:66]: Ah ok. We can iterate after.

**Pull request merged**